packages/contentful--react-apps-toolkit/src/SDKProvider.tsx

```diff
@@ -7,15 +7,35 @@ interface SDKProviderProps {
   loading?: ReactElement;
 }
 
+const DELAY_TIMEOUT = 4 * 1000;
+
 /**
  * The Component providing the AppSdk, the useSDK hook can only be used within this Provider
  * @param props.loading an optional loading element that gets rendered while initializing the app
  */
 export const SDKProvider: FC<SDKProviderProps> = (props) => {
   const [sdk, setSDK] = React.useState<KnownSDK | undefined>();
+  const timeout = useRef<number | undefined>();
+
+  const clearPossibleTimeout = () => {
+    if (timeout.current !== undefined) {
+      window.clearTimeout(timeout.current);
+    }
+  };
+
+  const onInitSuccess = (sdk: KnownSDK) => {
+    setSDK(sdk);
+    clearPossibleTimeout();
+  };
 
   React.useEffect(() => {
-    init(setSDK);
+    timeout.current = window.setTimeout(() => {
+      console.warn(
+        'Your app is taking longer than expected to initialize. If this is not fixed, let us know: https://github.com/contentful/ui-extensions-sdk/issues'
```

Comment:
The error in this case might still be in the user's code. What about rephrasing this making clear we won't do code reviews? :P 

```suggestion
        'Your app is taking longer than expected to initialize. If you think this is an error with Contentful's App SDK, let us know: https://github.com/contentful/ui-extensions-sdk/issues'
```